frontend/src/components/TimeLogsPage/TimeLogForm.js

```diff
@@ -20,10 +24,34 @@ export const TimeLogForm = ({ handleSubmit, disabled }) => {
     setDescription(event.target.value)
   }
 
+  const formIsInvalid = () => {
+    let error_exists = false
+    const timePattern = /^([0-1][0-9]|2[0-3]):[0-5][0-9]$/
+    if (!timePattern.test(time)) {
+      setFormTimeHasError(true)
+      setTimeErrorMessage('Time must be in format HH:MM')
+      error_exists = true
+    }
+    if (description.length < 5) {
+      setFormDescriptionHasError(true)
+      setDescriptionErrorMessage('Description must be at least 5 characters')
+      error_exists = true
+    }
+    return error_exists
+  }
+
   const handleFormSubmit = (event) => {
     event.preventDefault()
-    handleSubmit(date, time, description)
-    clearForm()
+    if (formIsInvalid()) {
```

Comment:
Maybe this can be removed too and just replace the else block with `if (!formIsInvalid())`?